posts/REDCapTidieR_v0.3.qmd

```diff
@@ -0,0 +1,103 @@
+
+For the above sample database, the typical `REDCapTidieR::read_readcap()` output stays the same:
+
+::: {.column-body-outset}
+![](images/repeat_event_db.png){fig-align="center" width="1000"}
+:::
+
+However, for the repeating instrument associated with the `r_instrument` form name, the new `redcap_form_instance` variable can be seen in the [red box]{style="color: red;"}:
+
+![](images/redcap_form_instance.png){fig-align="center" width="800"}
+
+And the new `redcap_event_instance` variable associated with the `nr_instrument` form name can be seen in the [green box]{style="color: green;"}:
+
+![](images/redcap_event_instance.png){fig-align="center" width="800"}
+
+### What this means for you
+
+The additional capabilities provided by [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [events](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event) support brought us back to the drawing board for how we wanted to best tackle REDCap [data tibbles](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#data-tibble). A cornerstone decision for REDCapTidieR's design is to remove any sparsity from a REDCap output, as given by the default [block matrix](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#block-matrix). By introducing `redcap_event_instance`, we are actively choosing to reintroduce the potential for `NA`/missing data. This rationale is based on the need to have [data tibble](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#data-tibble) [columns](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#data-column) preserved with singular meaning. Any one [column](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#data-column) should not be dependent on another for identification of [structure](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#structure). The [block matrix](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#block-matrix) and associated use of `redcap_repeat_instrument` and `redcap_repeat_instance` is not in line with the reduction in ambiguity that REDCapTidieR seeks to provide developers.
+
+This means that for longitudinal REDCap [databases](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#database), any use of `redcap_repeat_instance` needs to be updated to use the appropriate `redcap_form_instance`/`redcap_event_instance` variables.
```

Comment:
I would add that for the majority of developers who don't use repeating events all that needs to change is `redcap_repeat_instance` -> `redcap_form_instance` and they don't have to think about `redcap_event_instance`.

Reply:
Maybe even open this section with that info or something to the effect of "if you don't use repeating events all you have to do is make this one change"

Reply:
How does this sound for opening the section:

> For developers/maintainers of longitudinal REDCap's with no repeating events [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [events](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event), the only change to keep track of is the switch from `redca_repeat_instance` to `redcap_form_instance`.